app/src/main/java/es/wolfi/app/passman/FileViewAdapter.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Passman Android App
+ *
+ * @copyright Copyright (c) 2016, Sander Brand (brantje@gmail.com)
+ * @copyright Copyright (c) 2016, Marcos Zuriaga Miguel (wolfi@wolfi.es)
+ * @license GNU AGPL version 3 or any later version
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package es.wolfi.app.passman;
+
+import android.annotation.SuppressLint;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+import es.wolfi.passman.API.File;
+
+/**
+ * {@link RecyclerView.Adapter} that can display a {@link File} and makes a call to the
+ * specified {@link CredentialDisplay.OnListFragmentInteractionListener}.
+ */
+public class FileViewAdapter extends RecyclerView.Adapter<FileViewAdapter.ViewHolder> {
+
+    private final List<File> mValues;
+    private final CredentialDisplay.OnListFragmentInteractionListener filelistListener;
+
+    public FileViewAdapter(List<File> files, CredentialDisplay.OnListFragmentInteractionListener listener) {
+        mValues = files;
+        filelistListener = listener;
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.fragment_credential_file_item, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @SuppressLint("DefaultLocale")
+    public static String humanReadableByteCount(long bytes, boolean si) {
```

Comment:
This code is repeated across files, would be cool to extract it to a single function, we can leave this for a future version though

Reply:
Thanks to @axellebot for these changes in 38baaf1558bae1b4f54a9c53196b2543a62829b1